llvm/cmake/modules/HandleLLVMOptions.cmake

```diff
@@ -1064,7 +1064,8 @@ if(LLVM_USE_SANITIZER)
   if (LLVM_USE_SANITIZE_COVERAGE)
     append("-fsanitize=fuzzer-no-link" CMAKE_C_FLAGS CMAKE_CXX_FLAGS)
   endif()
-  if (LLVM_USE_SANITIZER MATCHES ".*Undefined.*")
+  if (LLVM_USE_SANITIZER MATCHES ".*Undefined.*" AND CMAKE_C_COMPILER_ID MATCHES "Clang" AND
+		                                     CMAKE_CXX_COMPILER_ID MATCHES "Clang")
```

Comment:
GCC supports `-fsanitize=undefined`: https://clang.godbolt.org/z/nT3bPW766

Maybe you have an old version that predates that support?

Reply:
Yes, but i don't touch `-fsanitize=undefined` argument, it passes both to clang and gcc. Here i disable `-fsanitize-blacklist=...` which doesn't supported by gcc

Reply:
ohh, I think it would help a lot to put that concept in a variable, e.g:

```
check_cxx_compiler_flag(-fsanitize-blacklist=${IGNORELIST_FILE} COMPILER_RT_CXX_SUPPORTS_FSANITIZE_BLACKLIST)
```

instead of tying this to the specific compiler.

Reply:
Thx, i've done the same for `-fno-sanitize=function` flag in llvm/CMakeLists.txt